### PR TITLE
Support Swift 6

### PR DESCRIPTION
**Pull request by 0xTim**
* Drop support for Swift 5.7
* Fix warnings in Vapor with latest Swift 6 changes
* Fix tests on Linux nightlies

### CODE REVIEW

File: Tests/VaporTests/ServerTests.swift [hunk 1]
```diff
@@ -42,6 +46,8 @@ final class ServerTests: XCTestCase, @unchecked Sendable {
         XCTAssertEqual(res.body?.string, "bar")
     }
     
+    // `httpUnixDomainSocket` is currently broken in 6.0
```
Comment (dimitribouniol) [inline Tests/VaporTests/ServerTests.swift:49]: Should we make a ticket for this and link it here?

File: Tests/VaporTests/URITests.swift [hunk 2]
```diff
@@ -183,13 +183,16 @@ final class URITests: XCTestCase {
         XCTAssertEqual(uri.string, "foobar://host:1/test?query#fragment")
     }
     
+    // Disable tests on Linux 6.0 until behaviour is fixed
+    #if compiler(<6.0)
```
Comment (dimitribouniol) [inline Tests/VaporTests/URITests.swift:187]: Same here

**Review by dimitribouniol — APPROVED** [review]
🚀

File: Tests/VaporTests/ServerTests.swift (hunk 1 above)
Comment (0xTim) [reply Tests/VaporTests/ServerTests.swift:49]: It's fixed in the 6.0 branch, just waiting on a release

File: Sources/Vapor/Logging/LoggingSystem+Environment.swift [hunk 3]
```diff
@@ -2,7 +2,8 @@ import Logging
 import ConsoleKit
 
 extension LoggingSystem {
-    public static func bootstrap(from environment: inout Environment, _ factory: (Logger.Level) -> (String) -> LogHandler) throws {
+    @preconcurrency
+    public static func bootstrap(from environment: inout Environment, _ factory: @Sendable (Logger.Level) -> (@Sendable (String) -> LogHandler)) throws {
```
Comment (dimitribouniol) [inline Sources/Vapor/Logging/LoggingSystem+Environment.swift:6]: Interesting that the API checker is flagging this as different — the parentheses are implied in the same spot on the old version, aren't they?

**Review by ptoffy — APPROVED** [review]

File: Sources/Vapor/Logging/LoggingSystem+Environment.swift (hunk 3 above)
Comment (0xTim) [reply Sources/Vapor/Logging/LoggingSystem+Environment.swift:6]: Yeah but I don't think the API checker is clever enough to work that out

**Pull request merged**